src/pages/index.js

```diff
@@ -125,21 +162,86 @@ function keyHandler(evt) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Loading...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      console.log(data);
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
 }
 
+function handleDeleteCard(cardElement, cardId) {
+  console.log(cardElement);
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  openModal(deleteModal);
+}
+//  const deleteSubmitButton =
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(deleteSubmitButton, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch((err) => console.error("Failed to delete card", err))
+    .finally(() => {
+      setButtonText(deleteSubmitButton, false, "Delete", "Deleting...");
+    });
+}
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
 function handleCardSubmit(evt) {
   evt.preventDefault();
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
-  console.log(cardSubmitButton);
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
-  cardNameInput.value = "";
-  cardLinkInput.value = "";
+  setButtonText(cardSubmitButton, true, "Loading...");
+
+  api
+    .postCard(inputValues)
+    .then((cardData) => {
+      const cardEl = getCardElement(cardData);
+      cardsList.prepend(cardEl);
+      console.log(cardSubmitButton);
+      //disableButton(cardSubmitButton, settings);
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute. 


Please, import the `disableButton()` function from the `validation.js` file, to be able to disable the save buttons in `.then` blocks


```js

      disableButton(cardSubmitButton, settings);

```